StripeFinancialConnections/StripeFinancialConnections/Source/Common/HostController.swift

```diff
@@ -96,41 +99,31 @@ extension HostController: HostViewControllerDelegate {
     ) {
         delegate?.hostController(self, didReceiveEvent: FinancialConnectionsEvent(name: .open))
 
-        if synchronizePayload.manifest.isProductInstantDebits {
-            continueWithWebFlow(synchronizePayload.manifest)
-        } else {
-            let flowRouter = FlowRouter(
-                synchronizePayload: synchronizePayload,
-                analyticsClient: analyticsClient
-            )
-            defer {
-                // no matter how we exit this function
-                // log exposure to one of the variants if appropriate.
-                flowRouter.logExposureIfNeeded()
-            }
-
-            guard flowRouter.shouldUseNative else {
-                continueWithWebFlow(synchronizePayload.manifest)
-                return
-            }
-
-            navigationController.configureAppearanceForNative()
-
-            let dataManager = NativeFlowAPIDataManager(
-                manifest: synchronizePayload.manifest,
-                visualUpdate: synchronizePayload.visual,
-                returnURL: returnURL,
-                consentPaneModel: synchronizePayload.text?.consentPane,
-                apiClient: api,
+        let flowRouter = FlowRouter(
+            synchronizePayload: synchronizePayload,
+            analyticsClient: analyticsClient
+        )
+
+        let flow = flowRouter.flow
+        analyticsClientV1.log(
+            analytic: FinancialConnectionsSheetFlowDetermined(
                 clientSecret: clientSecret,
-                analyticsClient: analyticsClient
-            )
-            nativeFlowController = NativeFlowController(
-                dataManager: dataManager,
-                navigationController: navigationController
-            )
-            nativeFlowController?.delegate = self
-            nativeFlowController?.startFlow()
+                flow: flow,
+                killswitchActive: flowRouter.killswitchActive
+            ),
+            apiClient: .shared
```

Comment:
I believe we can avoid the singleton, and pass in the variable as per precedent of passing in variables vs. singletons:

https://github.com/stripe/stripe-ios/pull/3763/files#diff-ab3536226ca6435b73fb87d20bbb229b3d6c948e2cf82207e31e2ee27aef39c6L124